apps/builder/app/builder/features/style-panel/sections/transitions/transition-content.tsx

```diff
@@ -43,34 +69,87 @@ export const TransitionContent = (props: TransitionContentProps) => {
       return;
     }
 
-    props.onEditLayer(props.index, layers);
+    onEditLayer(index, layers);
     setIntermediateValue(undefined);
   };
 
+  const handleOnPropertySelection = (newProperty: TupleValueItem) => {
+    const value: TupleValueItem[] = [
+      newProperty,
+      duration,
+      timing,
+      delay,
+    ].filter<TupleValueItem>((item): item is TupleValueItem => item !== null);
+
+    const newLayer: TupleValue = { type: "tuple", value };
+
+    setIntermediateValue({
+      type: "intermediate",
+      value: convertTupleToStringValue(newLayer),
+    });
+
+    onEditLayer(index, { type: "layers", value: [newLayer] });
+  };
+
   return (
-    <Flex
-      direction="column"
-      css={{
-        px: theme.spacing[5],
-        py: theme.spacing[5],
-        gap: theme.spacing[3],
-        minWidth: theme.spacing[30],
-      }}
-    >
-      <Label>Code</Label>
-      <TextArea
-        rows={3}
-        name="description"
-        css={{ minHeight: theme.spacing[14], ...textVariants.mono }}
-        value={intermediateValue?.value ?? props.value ?? ""}
-        onChange={handleChange}
-        onKeyDown={(event) => {
-          if (event.key === "Enter") {
-            handleComplete();
-            event.preventDefault();
-          }
+    <Flex direction="column">
+      <Grid
+        gap="2"
+        css={{
+          px: theme.spacing[8],
+          py: theme.spacing[8],
+          gridTemplateColumns: `1fr ${theme.spacing[22]}`,
+        }}
+      >
+        <TransitionProperty
+          /* Browser defaults for transition-property - all */
+          property={property ?? { type: "keyword", value: "all" }}
+          onPropertySelection={handleOnPropertySelection}
+        />
+      </Grid>
+      <Separator css={{ gridColumn: "span 2" }} />
+      <Flex
+        direction="column"
+        css={{
+          px: theme.spacing[8],
+          py: theme.spacing[8],
+          gap: theme.spacing[3],
+          minWidth: theme.spacing[30],
         }}
-      />
+      >
+        <Label>
+          <Flex align="center" gap="1">
+            Code
+            <Tooltip
+              variant="wrapped"
+              content={
+                <Text>
+                  Paste CSS code for a transition or part of a transition, for
+                  example:
+                  <br />
+                  <br />
+                  opacity 200ms ease;
```

Comment:
shouldn't this be a stringified version of the value we get as a default to sync these things

Reply:
We seem to use the same pattern in all the places. Buti tried `stip-indent`. It seems to collapse all spaces in jsx,